src/datasets/packaged_modules/spark/spark.py

```diff
@@ -85,6 +87,31 @@ def _info(self):
     def _split_generators(self, dl_manager: datasets.download.download_manager.DownloadManager):
         return [datasets.SplitGenerator(name=datasets.Split.TRAIN)]
 
+    def _repartition_df_if_needed(self, max_shard_size):
+        import pyspark
```

Comment:
Do we need to import pyspark inside the function?

Reply:
I don't think we need it since we imported in the ctor. Removed

Reply:
Never mind, tests failed with NameError so added it back